### PR TITLE
[libc] Breakup freelist_malloc into separate files

### PR DESCRIPTION
**Pull request by petrhosek**
This better matches the structure we use for the rest of libc.

### CODE REVIEW

**Review by PiJoules — APPROVED** [review]
LGTM

I think the build failure is from the targets being under `if(LLVM_LIBC_INCLUDE_SCUDO)` which I think should be false here.

**Comment by nickdesaulniers** [comment]
Has merge conflicts.

**Comment by SchrodingerZhu** [comment]
see also https://github.com/llvm/llvm-project/pull/119259

**Review by michaelrj-google — APPROVED** [review]
LGTM, thanks for the cleanup

**Comment by SchrodingerZhu** [comment]
@lntue this seems to be a better clean up

File: libc/src/__support/CMakeLists.txt [hunk 1]
```diff
@@ -48,20 +48,27 @@ add_header_library(
...
     .freestore
+    .freetrie
     libc.src.__support.CPP.cstddef
     libc.src.__support.CPP.array
     libc.src.__support.CPP.optional
     libc.src.__support.CPP.span
     libc.src.__support.libc_assert
     libc.src.string.memory_utils.inline_memcpy
     libc.src.string.memory_utils.inline_memset
+  COMPILE_OPTIONS
```
Comment (lntue) [inline libc/src/__support/CMakeLists.txt:69]: nit: can you move the `COMPILE_OPTIONS` part to above `DEPENDS`?

**Review by lntue — APPROVED** [review]

**Pull request merged**